python/cuml/kernel_ridge/kernel_ridge.pyx

```diff
@@ -0,0 +1,266 @@
+        corresponding kernel value as a single number.
+    gamma : float, default=None
+        Gamma parameter for the RBF, laplacian, polynomial, exponential chi2
+        and sigmoid kernels. Interpretation of the default value is left to
+        the kernel; see the documentation for sklearn.metrics.pairwise.
+        Ignored by other kernels.
+    degree : float, default=3
+        Degree of the polynomial kernel. Ignored by other kernels.
+    coef0 : float, default=1
+        Zero coefficient for polynomial and sigmoid kernels.
+        Ignored by other kernels.
+    kernel_params : mapping of str to any, default=None
+        Additional parameters (keyword arguments) for kernel function passed
+        as callable object.
+    Attributes
+    ----------
+    dual_coef_ : ndarray of shape (n_samples,) or (n_samples, n_targets)
+        Representation of weight vector(s) in kernel space
+    X_fit_ : ndarray of shape (n_samples, n_features)
+        Training data, which is also required for prediction. If
+        kernel == "precomputed" this is instead the precomputed
+        training matrix, of shape (n_samples, n_samples).
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        import cupy as cp
+        from cuml.kernel_ridge import KernelRidge
+        from numba import cuda
+        import math
+
+        n_samples, n_features = 10, 5
+        rng = cp.random.RandomState(0)
+        y = rng.randn(n_samples)
+        X = rng.randn(n_samples, n_features)
+
+        model = KernelRidge(kernel="poly").fit(X, y)
+        pred = model.predict(X)
+
+
+        @cuda.jit(device=True)
+        def custom_rbf_kernel(x, y, gamma=None):
+            if gamma is None:
+                gamma = 1.0 / len(x)
+            sum = 0.0
+            for i in range(len(x)):
+                sum += (x[i] - y[i]) ** 2
+            return math.exp(-gamma * sum)
+
+
+        model = KernelRidge(kernel=custom_rbf_kernel,
+                            kernel_params={"gamma": 2.0}).fit(X, y)
+        pred = model.predict(X)
+
+    """
+
+    dual_coef_ = CumlArrayDescriptor()
+
+    def __init__(
+        self,
+        *,
+        alpha=1,
+        kernel="linear",
+        gamma=None,
+        degree=3,
+        coef0=1,
+        kernel_params=None,
+        handle=None,
+        output_type=None,
+        verbose=False
+    ):
+        super().__init__(handle=handle, verbose=verbose,
+                         output_type=output_type)
+        self.alpha = cp.asarray(alpha)
+        self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
+        self.kernel_params = kernel_params
+
+    def _get_kernel(self, X, Y=None):
+        if isinstance(self.kernel, str):
+            params = {"gamma": self.gamma,
+                      "degree": self.degree, "coef0": self.coef0}
+        else:
+            params = self.kernel_params or {}
+        return pairwise_kernels(X, metric=self.kernel,
```

Comment:
`Y` is missing on this call to pairwise_kernels?

Reply:
Good catch. Tests are only passing because I'm predicting on the same X matrix as fit. Will fix.